### PR TITLE
Fixup some glib/gtk vala FTBFS

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Found a couple stragglers.

@romildo It appears that `deepin-mutter` and `deepin-wm` are not defunct upstream.
And `deepin-mutter` currently fails to build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jtojnar** [comment]
While at it, can you rename the font-manager homepage to https://github.com/FontManager/font-manager

**Comment by worldofpeace** [comment]
Here's the issue with deepin-terminal
* https://github.com/linuxdeepin/deepin-terminal/issues/193

Wasn't sure if that was the upstream xcb.vapi that had problems.

File: pkgs/applications/misc/font-manager/default.nix [hunk 1]
```diff
@@ -42,6 +42,10 @@ stdenv.mkDerivation rec {
     "-Ddisable_pycompile=true"
   ];
 
+  patches = [
+    ./fix-build-vala-0.46.patch
```
Comment (jtojnar) [inline pkgs/applications/misc/font-manager/default.nix]:
```suggestion
    # fix build with Vala 0.46
    (fetchpatch {
      url = "https://github.com/FontManager/font-manager/commit/c73b40de11f376f4515a0edfe97fb3721a264b35.patch";
      sha256 = "0lacwsifgvda2r3z6j2a0svdqr6mgav7zkvih35xa8155y8wfpnw";
      excludes = [ "fedora/font-manager.spec" ];
    })
```

Comment (worldofpeace) [reply pkgs/applications/misc/font-manager/default.nix]: Yep, forgot about that useful functionality :+1:

**Comment by worldofpeace** [comment]
> While at it, can you rename the font-manager homepage to https://github.com/FontManager/font-manager

We're actually using the pointless `repositories.git` attribute here.
And the actual homepage has a convenient link to the source repo.

I think I'll just remove the `repositories.git`.

**Review by hedning — APPROVED** [review]
LGTM

**Pull request merged**